src/aspire/source/relion.py

```diff
@@ -209,34 +209,38 @@ def _images(self, indices):
         # Log the indices in case needed to debug a crash
         logger.debug(f"Indices: {indices}")
 
-        def load_single_mrcs(filepath, df):
+        def load_single_mrcs(filepath, indices):
             arr = mrcfile.open(filepath).data
             # if the stack only contains one image, arr will have shape (resolution, resolution)
             # the code below reshapes it to (1, resolution, resolution)
             if len(arr.shape) == 2:
                 arr = arr.reshape((1,) + arr.shape)
             # __mrc_index is the 1-based index of the particle in the stack
-            data = arr[df["__mrc_index"] - 1, :, :]
+            data = arr[self._metadata["__mrc_index"][indices] - 1, :, :]
 
-            return df.index, data
+            return indices, data
```

Comment:
Since `indices` is now supplied as an argument, we probably don't need to return it anymore.

Reply:
I think it is actually unpacked in the future, and would be needed then , unless we keep track of the indices in another structure.
Good idea, but i'm not willing to touch that threadpool code in a rush. (it has no reason to exist, but that is another matter entirely).  Will create an issue to delete it.

Reply:
https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/922

Reply:
Yeah I saw that, so issue is probably the way to go. Thanks.